pkg/kubectl/stop.go

```diff
@@ -270,10 +271,70 @@ func (reaper *JobReaper) Stop(namespace, name string, timeout time.Duration, gra
 		return utilerrors.NewAggregate(errList)
 	}
 	// once we have all the pods removed we can safely remove the job itself
-	if err := jobs.Delete(name, gracePeriod); err != nil {
+	return jobs.Delete(name, gracePeriod)
+}
+
+func (reaper *DeploymentReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *api.DeleteOptions) error {
+	deployments := reaper.Extensions().Deployments(namespace)
+	replicationControllers := reaper.ReplicationControllers(namespace)
+
+	deployment, err := deployments.Get(name)
+	if err != nil {
+		return err
+	}
+
+	// set deployment's history and scale to 0
+	zero := 0
+	deployment.Spec.RevisionHistoryLimit = &zero
+	deployment.Spec.Replicas = 0
+	deployment, err = deployments.Update(deployment)
+	if err != nil {
 		return err
 	}
-	return nil
+
+	// wait for total no of pods drop to 0
+	if err := wait.Poll(reaper.pollInterval, reaper.timeout, func() (bool, error) {
+		curr, err := deployments.Get(name)
+		if err != nil {
+			return false, nil
+		}
+		return curr.Status.Replicas == 0, nil
+	}); err != nil {
+		return err
+	}
+
+	// before removing RCs we need to pause the deployment, so that new RCs don't get created
+	deployment, err = deployments.Get(name)
+	if err != nil {
+		return err
+	}
+	deployment.Spec.Paused = true
+	deployment, err = deployments.Update(deployment)
+	if err != nil {
+		return err
+	}
+
+	// remove remaining RCs
```

Comment:
IIUC, we set Paused=true to stop the deployment from creating new RC. Perhaps in deployment controller, we shouldn't create new RC when deployment.Spec.Replicas=0 (currently it creates a rc with 0 replicas)?